Documenting usage of yaml configuration file

olelod opened the pull request:
*Insert a description of your pull request (PR) here, and check off the boxes below when they are done.*

---

### Contributor checklist

- [ ] :tada: This PR closes #ISSUE_NUMBER.
- [ ] :scroll: I have broken down my PR into the following tasks:
   - [ ] Task 1
   - [ ] Task 2
- [ ] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [ ] :book: I have considered adding a new entry in `CHANGELOG.md`.
- [ ] :books: I have considered updating the documentation.

olwijn commented:
A couple of small suggesstions for improvements:

- In the discussion on the bulk volume, please mention the name of the additional option for its distribution
- In the example of the usage of the relperm options, several entries are empty. is this correct? please explain. related to this: does the concept of default values apply anywhere in this discussion?
- In several locations the word scheme is misspelled shceme
- Some quantities in the relperm figures doe snot appear as input options (KRGMAX, KRWMAX). also, should I not expect an option for irreducable gas saturation, or is this assumed to be 0?
- The relperm interpolation  requires base value but I do not see this used in an example. in the reperm example table, maybe add a third column where interpolation is used
- The reason for and workings of the low_optimistic option is not entirely clear
- There are a couple of small typo's here and there. Can the text be run through a spelling checker perhaps?
- From the description it is not entirely clear why there is a delta_depth option (what is its impact?). Can this be explained here
- When the individual scheme is specified, does the size_in_bulkvolumes value apply to each aquifer individually?

olelod commented:
> A couple of small suggesstions for improvements:
> 
> * In the discussion on the bulk volume, please mention the name of the additional option for its distribution

The `prior_volume_distribution` described. Its in the flownet part of the config.

> * In the example of the usage of the relperm options, several entries are empty. is this correct? please explain. related to this: does the concept of default values apply anywhere in this discussion?

Should not be empty. Filled now.

> * In several locations the word scheme is misspelled shceme

Fixed

> * Some quantities in the relperm figures do not appear as input options (e.g. KRGMAX, KRWMAX). Also, should I not expect an option for irreducable gas saturation, or is this assumed to be 0?

These two were defaulted to 1. We added the krwmax a little while back (and the possibility for krwmax_add_to_krwend). The krgmax is still defaulted to 1, and not recognized as a keyword in the config parser. Don't think we need to add it, but maybe we should for completeness?

> * The relperm interpolation requires a base value but I do not see this used in an example. In the reperm example table, maybe add a third column where interpolation is used.

Added

> * Is it possible to add a figure that illustrates the interpolation approach?

I guess it would be possible, but haven't gotten around to it as of yet.

> * The reason for and workings of the low_optimistic option is not entirely clear

Tried to elaborate slightly more


> * There are a couple of small typo's here and there. Can the text be run through a spelling checker perhaps?

Yes, done

> * From the description it is not entirely clear why there is a delta_depth option (what is its impact?). Can this be explained here
> * When the individual scheme is specified, does the size_in_bulkvolumes value apply to each aquifer individually?

olelod commented:
It's maybe easier to read here: https://olelod.github.io/flownet/configuration_file.html

I update the gh-pages on my fork as I update the configuration_file.rst, so they should be equal.

augustjohansson commented on docs/configuration_file.rst:
```diff
@@ -1,22 +1,1167 @@
+These keywords are mainly related to the geometry of the model, populating with model parameters will be described under 
+`model_parameters`_. For simplicity the keywords are listed alphabetically (by indentation), they can appear in any order 
+in the actual configuration file.
+
```
Should it be mentioned which unit convention is used? For example, there are pressure inputs for a few options, as well as the `min_permeability` option (although here it is explicitly said that it should be in mD).

olelod replied:
I added information on how the unit conversion can potentially be changed. Don't think it should be necessary with a full list of units.

augustjohansson commented on docs/configuration_file.rst:
```diff
@@ -1,22 +1,1167 @@
+fast_pyscal
+-----------
+
+maybe not relevant anymore?
```
I did a look in the code and it does not seem to be used. I can create a pull request removing this option if you'd like.

olelod replied:
We used the fast option in pyscal earlier, but changed to our own numpy implementation. We still use pyscal in the tests to see that our numpy implementation and pyscal generate the same SWOF/SGOF tables. I have removed the reference to pyscal in the documentation, but i guess we can just leave it in `_config_parser.py` for now.

augustjohansson commented on docs/configuration_file.rst:
```diff
@@ -1,22 +1,1167 @@
+|  flownet:                        |  flownet:                        |
+|    model_parameters:             |    model_parameters:             |
+|      rock_compressibility:       |      rock_compressibility:       |
+|        reference_pressure:       |        reference_pressure:       |
```
As an example, the Brugge model has
  rock_compressibility:
```
    reference_pressure: 170
    min: 6.0e-6
    max: 5.0e-5
```

augustjohansson commented on docs/configuration_file.rst:
```diff
@@ -1,22 +1,1167 @@
+hyperopt
+--------
+
+A dictionary with parameters relater to hyper optimization of input.
```
Spelling

augustjohansson replied:
Maybe it'd help with a reference to what this involves?

augustjohansson commented on docs/configuration_file.rst:
```diff
@@ -1,22 +1,1167 @@
+concave_hull
+~~~~~~~~~~~~
+
+ADD MORE HERE
```
In `_config_parser.py` it is stated that 
"When true, the bounding boxes of the gridcells of the original reservoir model are used to check if the generated additional nodes are positioned within the reservoir volume."
I think this is good enough (but I haven't checked if it's correct or not!)

augustjohansson commented on docs/configuration_file.rst:
```diff
@@ -1,22 +1,1167 @@
+flownet
+=======
+
+A description of the usage of the various keywords available in relation to the construction of the FlowNet network model. 
```
Perhaps there should be a reference to `_config_parser.py` for default parameters? Maybe that is better than writing the  default values of every parameter in this document.

olelod replied:
I guess we should have a better way of listing default values - more consistent between the different keywords, but I don't think we should ask the user to go into the code to find them. I'm sure we will figure something out.

augustjohansson commented on docs/configuration_file.rst:
```diff
@@ -1,22 +1,1167 @@
+Valid options are:
+
+* tube_length: distrubutes the volume of the convex hull of the FlowNet model,
+  based on the length of a tube. I.e., if all tubes have equeal lenght, they
```
spelling equeal lenght

augustjohansson commented on docs/configuration_file.rst:
```diff
@@ -1,22 +1,1167 @@
+
+Defining this at the same time as **training_set_end_date** will raise a ValueError.
+      
+Example of the entire flownet part of the configuration yaml file:
```
Perhaps this complete example could be moved to the very top, in the introduction to flownet?

augustjohansson commented on docs/configuration_file.rst:
```diff
@@ -1,22 +1,1167 @@
+realizations
+------------
+
+A dictionary with some key/value pairs that control the number of realizations to submit to ERT, and how these 
```
Is it sufficient to be this vague?

augustjohansson reviewed (commented):
Here are my minor comments on the documentation.

FrancescaWatson commented on docs/configuration_file.rst:
```diff
@@ -1,22 +1,1167 @@
+a more optimistic relative permeability curve. This can be indicated by setting 
+**low_optimistic** to **True** for that parameter.
+
+A parameter value on the interval [-1,0) will interpolate all input parameters 
```
I'm a bit confused but it might just be because I don't know enough about the topic. Does the value of the interpolation parameter make a difference or would any number in e.g. [-1,0) have the same effect?

olelod replied:
It's a history matching parameter, -1 will reproduce the low case, 0 will reproduce the base case, 1 will reproduce the high case. All other numbers will be linear interpolations between the inputs in the three defined cases.

FrancescaWatson reviewed (commented):
I put in a few comments where I didn't understand things and noted some typos.

olelod commented on docs/configuration_file.rst:
```diff
@@ -1,22 +1,1167 @@
-The configuration files follows the `YAML standard <https://yaml.org/>`_.
+The configuration file contains all information needed to build, run and analyse the results from the FlowNet network model.
+
+name
```
It's in the config_parser.py so ill leave it here as well. Added a little text.

fahaddilib reviewed (approved):
👍 great to get this merged
